common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -108,7 +108,7 @@ impl NavigationController {
                         let current_geometry_index = index_of_closest_origin_point(
                             *snapped_user_location,
                             &self.route.get_linestring(),
-                            *current_geometry_index,
+                            current_geometry_index.unwrap_or(0),
```

Comment:
Do we always want to fall back to zero? Given the fact that we can "skip" forward and it's impossible to go backward, is it possible that we could lock to the final index, skipping the entire array?

I am still not 100% sure if this is the behavior we want (you've probably thought it through better than me, so maybe you can explain if I'm missing something?), but if it is, we need to consider the ramifications and probably have the function either return a Result with a descriptive Error type or an enum describing the different states.

If we don't do this, then I _think_ the result should always be Some, so we should log what happened (at a minimum) so we can detect bugs.